sp800-63c/sec5_assertions.md

```diff
@@ -79,13 +83,17 @@ Assertions SHALL use audience restriction techniques to allow an RP to recognize
 
 In some circumstances, it is desirable to prevent the subscriber's account at the IdP from being easily linked at multiple RPs through use of a common identifier. In these circumstances, pairwise pseudonymous subject identifiers SHALL be used within the assertions generated by the IdP for the RP, and the IdP SHALL generate a different identifier for each RP as described in [Section 5.2.6](#ppi-gen) below.
 
+>**MG: Any concern that "in some circumstances" with a SHALL is too weak?**
+
 When pairwise pseudonymous identifiers are used with RPs alongside attributes, it may still be possible for multiple colluding RPs to reidentify a subscriber by correlation across systems using these identity attributes. For example, if two independent RPs each see the same subscriber identified with different pairwise pseudonymous identifiers, they could still determine that the subscriber is the same person by comparing the name, email address, physical address, or other identifying attributes carried alongside the pairwise pseudonymous identifier in the respective assertions. Privacy policies MAY prohibit such correlation, and pairwise pseudonymous identifiers can increase effectiveness of these policies by increasing the administrative effort in managing the attribute correlation. 
 
 Note that in a proxied federation model, the initial IdP may be unable to generate a pairwise pseudonymous identifier for the ultimate RP, since the proxy could blind the IdP from knowing which RP is being accessed by the subscriber. In such situations, the pairwise pseudonymous identifier is generally established between the IdP and the federation proxy itself. The proxy, acting as an IdP, can itself provide pairwise pseudonymous identifiers to downstream RPs. Depending on the protocol, the federation proxy may need to map the pairwise pseudonymous identifiers back to the associated identifiers from upstream IdPs in order to allow the identity protocol to function. In such cases, the proxy will be able to track and determine which pairwise pseudonymous identifiers represent the same subscriber at different RPs.
 
+>**MG: Can we at least provide a restriction here on reuse of that mapping?**
+
 #### 5.2.6. <a name="ppi-gen"></a>Pairwise Pseudonymous Identifier Generation
 
-Pairwise pseudonymous identifiers SHALL be opaque and unguessable, containing no identifying information about the subscriber. Normally, the identifiers SHALL only be known by and used by one IdP-RP pair. However, an IdP MAY generate the same identifier for a subscriber at multiple RPs at the request of those RPs, but only if:
+Pairwise pseudonymous identifiers SHALL be opaque and unguessable, neither containing nor generated from identifying information about the subscriber. Normally, the identifiers SHALL only be known by and used by one IdP-RP pair or IdP-Proxy and Proxy-RP pair. However, an IdP MAY generate the same identifier for a subscriber at multiple RPs at the request of those RPs, but only if:
```

Comment:
To be clear, there are two issues in this line. The first is the mention of proxies, which we should remove entirely. 

The second is the restriction on generation of the PPI. We basically don't want it to include or be easily guessable from privacy-sensitive attributes like names or usernames or employee numbers or what have you. You can generate the PPI from those attributes but only if you use some kind of deterministic keyed hash, as Jim suggests. I tried to spell out the guidelines in the bulleted list above. but to Jim's point we're more interested in making sure people don't do something stupid than we are in telling people exactly how to do it. As a compromise, let's have a couple of prohibitions (can't contain identifying information directly, can't use an unkeyed hash, can't be an easily guessable serial number), and then have a SHOULD for guidance of what you can do instead but leave the door open for other reasonable implementations.